packages/next/shared/lib/utils.ts

```diff
@@ -1,13 +1,13 @@
-import { IncomingMessage, ServerResponse } from 'http'
-import { ParsedUrlQuery } from 'querystring'
-import { ComponentType } from 'react'
-import { UrlObject } from 'url'
 import { formatUrl } from './router/utils/format-url'
-import { NextRouter } from './router/router'
-import { Env } from '@next/env'
-import { BuildManifest } from '../../server/get-page-files'
-import { DomainLocales } from '../../server/config'
-import { PreviewData } from 'next/types'
+import type { BuildManifest } from '../../server/get-page-files'
```

Comment:
We've been holding off on using `import type` to prevent breaking users on older TypeScript versions, we might be able to leverage this now that we have a warning for this although might be better to wait for the next minor release

Reply:
Actually looks like we already started using this syntax [here](https://github.com/vercel/next.js/blob/bdcb8b0a716601fe366beab9ecdc90eb3df2b7ea/packages/next/client/router.ts#L4) so should be alright